### ISSUE TITLE
In a separate project, a namespace sharing the same name as a module with ModuleSuffix is not visible

### ISSUE THREAD

**Issue by mciccottidmg** [dotnet/fsharp#14798]
If a project with a module with the same name as a namespace occur in the same project (but different files), only the things in the module will be visible in other projects.

Steps to reproduce the problem:

1. Create a module with a value in it with the ModuleSuffix attribute
```fsharp
[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
module NamespaceModuleCollision.TopLevel

let test = 5
```
2. Create a second module inside a namespace with the same name as the above module, also with a value in it
```fsharp
module NamespaceModuleCollision.TopLevel.Sub

let test2 = 5
```
3. In a different project, reference the first project and attempt to use the values from both modules
```fsharp
let _ = NamespaceModuleCollision.TopLevel.Sub.test2 //Compiles fine

let _ = NamespaceModuleCollision.TopLevel.test //fails to compile

open NamespaceModuleCollision.TopLevel
let _  = test //fails to compile
```

[NamespaceModuleCollision.zip](https://github.com/dotnet/fsharp/files/10818194/NamespaceModuleCollision.zip)

**Expected behavior**

The values in the NamespaceModuleCollision.TopLevel can be referenced.

**Actual behavior**

The values in the NamespaceModuleCollision.TopLevel module result in compiler errors:

error FS0039: The value, constructor, namespace or type 'test' is not defined.
error FS0039: The value or constructor 'test' is not defined. Maybe you want one of the following:↔   Text

**Known workarounds**

Name the module and namespace different things.

**Related information**
Reproduced in a net7.0 project

**Comment by vzarytovskii** [comment]
I have tested it with different frameworks, as old as netcoreapp3.1 and net472, and it still produces the same issue, so if it's a regression, it's for sure not a recent one.

Not entirely sure it ever worked, @nojaf, @auduchinok, @dsyme thoughts? @mciccottidmg did it ever work for you?

**Comment by mciccottidmg** [comment]
@vzarytovskii No, I don't think it's a regression. Similar to my comment on https://github.com/dotnet/fsharp/issues/14799, it works within a single project if the namespace and module are declared in different files. I would assume that should also then work across project boundaries.

**Comment by vzarytovskii** [comment]
Cross project (i.e. cross-assembly) is different, must be in the spec how do we handle it.
I'll convert it into suggestion.

**Comment by xperiandri** [comment]
So you mean that to fix that first we need to describe the solution and approve it?

**Comment by Martin521** [comment]
The `CompilationRepresentation` attribute changes the .net name of the module (i.e. class), but not the F# name. The F# compiler should use it to convert between these names when importing or writing the assembly.
However, for all syntactic and checking purposes (including shadowing), only the F# name should matter.
This means that the above example should lead to `error FS0247: A namespace and a module named 'NamespaceModuleCollision.TopLevel' both occur in two parts of this assembly` in the same way as it does without the attribute.
(And then the question of referencing the modules does not even come up. And if somebody would be able to create such an assembly by other means, F# should throw and error while importing it.)
